src/Infrastructure/Violation/Formatter/ViolationReportDecorator.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+declare(strict_types=1);
+
+namespace SprykerSdk\Sdk\Infrastructure\Violation\Formatter;
+
+use SprykerSdk\Sdk\Core\Application\Dto\Violation\PackageViolationReport;
+use SprykerSdk\Sdk\Core\Application\Dto\Violation\ViolationReport;
+use SprykerSdk\SdkContracts\Report\Violation\ViolationInterface;
+use SprykerSdk\SdkContracts\Report\Violation\ViolationReportInterface;
+
+class ViolationReportDecorator
+{
+    /**
+     * @var iterable<\SprykerSdk\Sdk\Infrastructure\Violation\Formatter\OutputViolationDecoratorInterface>
+     */
+    protected iterable $violationDecorators;
+
+    /**
+     * @param iterable<\SprykerSdk\Sdk\Infrastructure\Violation\Formatter\OutputViolationDecoratorInterface> $violationDecorators
+     */
+    public function __construct(iterable $violationDecorators)
+    {
+        $this->violationDecorators = $violationDecorators;
+    }
+
+    /**
+     * @param \SprykerSdk\SdkContracts\Report\Violation\ViolationReportInterface $violationReport
+     *
+     * @return \SprykerSdk\SdkContracts\Report\Violation\ViolationReportInterface
+     */
+    public function decorate(ViolationReportInterface $violationReport): ViolationReportInterface
+    {
+        $violations = [];
+
+        foreach ($violationReport->getViolations() as $violation) {
+            $violations[] = $this->decorateViolation($violation);
+        }
+
+        $packages = [];
+
+        foreach ($violationReport->getPackages() as $package) {
+            $packagesViolations = [];
+            foreach ($package->getViolations() as $violation) {
+                $packagesViolations[] = $this->decorateViolation($violation);
+            }
+
+            $packageFileViolations = [];
+            foreach ($package->getFileViolations() as $path => $fileViolations) {
```

Comment:
Please, extract this `foreach` loop to the separate method. It is hard to read/understand, imo.